09_employment/employment_metrics_county_all.qmd

```diff
@@ -0,0 +1,856 @@
+---
+title: "Employment Opportunities Predictor, County-level"
+author: "JP Walsh and Aaron R. Williams"
+date: today
+format:
+  html:
+    toc: true
+    toc_float: true
+    embed-resources: true
+    code-fold: show
+execute: 
+  warning: false
+editor_options: 
+  chunk_output_type: console
+---
+
+*Program for the creation of the Employment Opportunities Upward Mobility Predictor at the County-level*
+
+ACS Code: Initially written by Tina Chelidze and updated by JP Walsh 2022-2023.
+
+Primary data derived from the IPUMS API.
+
+Based on processes developed by Paul Johnson and Kevin Werner in SAS.
+
+## Housekeeping
+
+Read in packages for use. If you do not have an IPUMS API key you will need to make one via the [IPUMS website](https://www.ipums.org/).
+
+```{r}
+library(tidyverse)
+library(ipumsr)
+library(reactable)
+library(srvyr)
+library(scales)
+library(furrr)
+library(tictoc)
+
+options(scipen = 999)
+
+theme_set(theme_minimal())
+
+# DO NOT PUSH YOUR API KEY. You only have to run this once and then comment it out as below.
+#set_ipums_api_key("Your KEY", save = TRUE)
+
+purrr::walk(list.files(here::here("functions", "API")), 
+    ~source(here::here("functions", "API", .x)))
+
+```
+
+
+Read in custom functions for this program. 
+```{r}
+
+calc_survey_mean <- function(.data) {
+  
+  .data %>%
+    as_survey_rep(
+      weights = perwt, 
+      repweights = matches("repwtp[0-9]+"),
+      type = "JK1",
+      scale = 4 / 80,
+      rscales = rep(1, 80), 
+      mse = TRUE
+    ) %>% 
+    group_by(year, crosswalk_period, statefip, county) %>% 
+    summarise(
+      share_employed = survey_mean(employed, vartype = "ci"),
+      n = n()
+    ) %>%
+    ungroup()
+  
+}
+
+finalize_metric <- function(.data) {
+  
+  .data %>%
+    arrange(year, statefip, county) %>%
+    mutate(
+      share_employed_lb = pmax(share_employed_low, 0),
+      share_employed_ub = pmin(share_employed_upp, 1)
+    ) %>%
+    select(-share_employed_low, -share_employed_upp, -n)
+  
+}
+```
+
+## Read ACS Data
+
+### Person-Level Data
+
+Read in the ACS extracts using the `extract_ipums()` function. Make sure to change the survey list to reflect what years you want to include in the data. Currently we are including three years of 1-year data (2018a, 2021a, and 2022a) for the overall calculation.
+
+```{r}
+acs <- extract_ipums(
+  extract_name = "umf_data_18_22",
+  extract_description = "Microdata pull for Mobility Metric Predictors.
+  American Community Survey, overall, years 2018, 2019 and 2022 (1-year).",
+  survey = list("us2018a", "us2021a", "us2022a")
+)
+
+```
+
+```{r}
+acs <- acs %>%
+  filter(sample %in% c("2018 ACS", "2021 ACS", "2022 ACS"))
+
+```
+
+Look at the distribution of survey samples in the data. The number of unique samples in the data should match the number of surveys selected in the `extract_ipums()` function above.
```

Comment:
I worry about this because the sample selected will change as the code is updated so I don't want to hard code the test. The idea with the count is just to give the user a spot to see what surveys are read in.